src/types.ts

```diff
@@ -172,6 +172,17 @@ export interface KeyingConfig {
   [typename: string]: KeyGenerator;
 }
 
+export type SerializedEntry = EntityField | Connection[] | Link;
+
+export interface SerializedEntries {
+  [key: string]: SerializedEntry;
+}
+
+export interface StorageAdapter {
+  read(): Promise<SerializedEntries>;
```

Comment:
Problem being that we'll have to change the implementation from doing `.then` to 

if (.then) x else y

Reply:
Yea, I think it’s not worth adding this since results can just be wrapped in `Promise.resolve()`. Either way, I think we do want to encourage non-synchronous storage options

Reply:
> we do want to encourage non-synchronous storage options

I thought this was the opposite! Thanks for clearing this up